engine/src/flutter/lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -679,6 +702,24 @@ abstract class SemanticRole {
     }
   }
 
+  void _updateControls() {
```

Comment:
This code needs a test.

Reply:
yup i will add a test, was waiting for confirmation this is the approach we want to take